tunnels.go

```diff
@@ -0,0 +1,294 @@
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/cfd_tunnel", params.AccountID)
+
+	tunnel := Tunnel{Name: params.Name, Secret: params.Secret}
+
+	res, err := api.makeRequestContextWithHeaders(ctx, http.MethodPost, uri, tunnel, tunnelV1Header())
+	if err != nil {
+		return Tunnel{}, err
+	}
+
+	var argoDetailsResponse TunnelDetailResponse
+	err = json.Unmarshal(res, &argoDetailsResponse)
+	if err != nil {
+		return Tunnel{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return argoDetailsResponse.Result, nil
+}
+
+// UpdateTunnel updates an existing tunnel for the account.
+//
+// API reference: https://api.cloudflare.com/#cloudflare-tunnel-update-cloudflare-tunnel
+func (api *API) UpdateTunnel(ctx context.Context, params TunnelUpdateParams) (Tunnel, error) {
+	if params.AccountID == "" {
+		return Tunnel{}, ErrMissingAccountID
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/cfd_tunnel", params.AccountID)
+
+	var tunnel Tunnel
+
+	if params.Name != "" {
+		tunnel.Name = params.Name
+	}
+
+	if params.Secret != "" {
+		tunnel.Secret = params.Secret
+	}
+
+	res, err := api.makeRequestContextWithHeaders(ctx, http.MethodPatch, uri, tunnel, tunnelV1Header())
+	if err != nil {
+		return Tunnel{}, err
+	}
+
+	var argoDetailsResponse TunnelDetailResponse
+	err = json.Unmarshal(res, &argoDetailsResponse)
+	if err != nil {
+		return Tunnel{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return argoDetailsResponse.Result, nil
+}
+
+// DeleteTunnel removes a single Argo tunnel.
+//
+// API reference: https://api.cloudflare.com/#cloudflare-tunnel-delete-cloudflare-tunnel
+func (api *API) DeleteTunnel(ctx context.Context, params TunnelDeleteParams) error {
+	uri := fmt.Sprintf("/accounts/%s/cfd_tunnel/%s", params.AccountID, params.UUID)
+
+	res, err := api.makeRequestContextWithHeaders(ctx, http.MethodDelete, uri, nil, tunnelV1Header())
+	if err != nil {
+		return err
+	}
+
+	var argoDetailsResponse TunnelDetailResponse
+	err = json.Unmarshal(res, &argoDetailsResponse)
+	if err != nil {
+		return errors.Wrap(err, errUnmarshalError)
+	}
+
+	return nil
+}
+
+// CleanupTunnelConnections deletes any inactive connections on a tunnel.
+//
+// API reference: https://api.cloudflare.com/#cloudflare-tunnel-clean-up-cloudflare-tunnel-connections
+func (api *API) CleanupTunnelConnections(ctx context.Context, params TunnelCleanupParams) error {
+	if params.AccountID == "" {
+		return ErrMissingAccountID
+	}
+
+	if params.UUID == "" {
+		return errors.New("missing tunnel UUID")
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/cfd_tunnel/%s/connections", params.AccountID, params.UUID)
+
+	res, err := api.makeRequestContextWithHeaders(ctx, http.MethodDelete, uri, nil, tunnelV1Header())
+	if err != nil {
+		return err
+	}
+
+	var argoDetailsResponse TunnelDetailResponse
+	err = json.Unmarshal(res, &argoDetailsResponse)
+	if err != nil {
+		return errors.Wrap(err, errUnmarshalError)
+	}
+
+	return nil
+}
+
+// TunnelToken gets the cloudflared token for a tunnel.
```

Comment:
token that allows to run a tunnel

Reply:
done! [258d853](https://github.com/cloudflare/cloudflare-go/pull/866/commits/258d85302dca121e71c062e18e28eeb35b474519)